app/code/Magento/Catalog/Test/Unit/Block/Product/View/GalleryOptionsTest.php

```diff
@@ -0,0 +1,278 @@
+     * @var View|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $configView;
+
+    /**
+     * @var Config|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $viewConfig;
+
+    /**
+     * @var Escaper
+     */
+    private $escaper;
+
+    protected function setUp()
+    {
+        $objectManager = new ObjectManager($this);
+
+        $this->escaper = $objectManager->getObject(Escaper::class);
+        $this->configView = $this->createMock(View::class);
+
+        $this->viewConfig = $this->createConfiguredMock(
+            Config::class,
+            [
+                'getViewConfig' => $this->configView
+            ]
+        );
+
+        $this->context = $this->createConfiguredMock(
+            Context::class,
+            [
+                'getEscaper' => $this->escaper,
+                'getViewConfig' => $this->viewConfig
+            ]
+        );
+
+        $this->gallery = $this->createMock(Gallery::class);
+
+        $this->jsonSerializer = $objectManager->getObject(
+            Json::class
+        );
+
+        $this->model =  $objectManager->getObject(GalleryOptions::class, [
+            'context' => $this->context,
+            'jsonSerializer' => $this->jsonSerializer,
+            'gallery' => $this->gallery
+        ]);
+    }
+
+    public function testGetOptionsJson()
+    {
+        $configMap = [
+            ['Magento_Catalog', 'gallery/nav', 'thumbs'],
+            ['Magento_Catalog', 'gallery/loop', false],
+            ['Magento_Catalog', 'gallery/keyboard', true],
+            ['Magento_Catalog', 'gallery/arrows', true],
+            ['Magento_Catalog', 'gallery/caption', false],
+            ['Magento_Catalog', 'gallery/allowfullscreen', true],
+            ['Magento_Catalog', 'gallery/navdir', 'horizontal'],
+            ['Magento_Catalog', 'gallery/navarrows', true],
+            ['Magento_Catalog', 'gallery/navtype', 'slides'],
+            ['Magento_Catalog', 'gallery/thumbmargin', '5'],
+            ['Magento_Catalog', 'gallery/transition/effect', 'slide'],
+            ['Magento_Catalog', 'gallery/transition/duration', '500'],
+        ];
+
+        $imageAttributesMap = [
+            ['product_page_image_medium','height',null, 100],
+            ['product_page_image_medium','width',null, 200],
+            ['product_page_image_small','height',null, 300],
+            ['product_page_image_small','width',null, 400]
+        ];
+
+        $this->configView->expects($this->any())
+            ->method('getVarValue')
+            ->will($this->returnValueMap($configMap));
+        $this->gallery->expects($this->any())
+            ->method('getImageAttribute')
+            ->will($this->returnValueMap($imageAttributesMap));
+
+        $json = $this->model->getOptionsJson();
+
+        $decodedJson = $this->jsonSerializer->unserialize($json);
+
+        $this->assertEquals('thumbs', $decodedJson['nav']);
+        $this->assertInternalType("string", $decodedJson['nav']);

```

Comment:
Please remove all such assertions. Simply use `assertSame` instead of `assertEquals`.

Reply:
@gwharton by using `assertSame` all `assertInternalType` may be removed - that's what I meant ;)

Reply:
OK, i'm with you. :)